foucaultview.cpp

```diff
@@ -152,6 +169,328 @@ QVector<QPoint> scaleProfile(QPolygonF points, int width,
+                if (k_side > hx) mask.at<double>(y, x) = 255.0;
+            }
+        }
+    }
+
+    // 4. DFT Pipeline
+    cv::Mat FFT1, FFT2, complexMask, complexSlit, combinedFilter, finalResult;
+    cv::Mat planesM[] = {mask, cv::Mat::zeros(mask.size(), CV_64FC1)};
+    cv::Mat planesS[] = {sourceSlit, cv::Mat::zeros(sourceSlit.size(), CV_64FC1)};
+    cv::merge(planesM, 2, complexMask);
+    cv::merge(planesS, 2, complexSlit);
+
+    cv::dft(complexMask, FFT1, cv::DFT_REAL_OUTPUT);
+    cv::dft(complexSlit, FFT2, cv::DFT_REAL_OUTPUT);
+    if (type == OpticalTestType::Ronchi) { shiftDFT(FFT1); shiftDFT(FFT2); }
+
+    cv::mulSpectrums(FFT1, FFT2, combinedFilter, 0, true);
+    cv::idft(combinedFilter, combinedFilter, cv::DFT_SCALE);
+    if (type == OpticalTestType::Ronchi) shiftDFT(combinedFilter);
+
+    cv::mulSpectrums(combinedFilter, surf_fft, finalResult, 0, true);
+    cv::idft(finalResult, finalResult, cv::DFT_SCALE);
+    if (type == OpticalTestType::Ronchi) shiftDFT(finalResult);
+
+    // 5. Output Image
+    QImage res = showMag(finalResult, false, "", false, s.gamma);
+    int startx = size - wf->data.cols;
+    return res.copy(startx, startx, wf->data.cols, wf->data.cols).mirrored(true, false);
+}
+
+
+void foucaultView::on_makePb_clicked()
+{
+    m_guiTimer.stop();
+    if (m_wf == nullptr || m_wf->data.cols == 0)
+        return;
+
+    if (mirrorDlg::get_Instance()->isEllipse()){
+        QMessageBox::warning(0,"warning","Foucaualt is not suppported for flat surfaces");
+        return;
+    }
+
+    QApplication::setOverrideCursor(Qt::WaitCursor);
+
+    // 1. Pack the current UI state into the settings struct
+    OpticalTestSettings settings;
+    settings.rocOffset    = ui->rocOffsetSb->value();
+    settings.ronchiX      = ui->RonchiX->value();
+    settings.lpi          = ui->lpiSb->value();
+    settings.gamma        = ui->gammaSb->value();
+    settings.slitWidth    = ui->slitWidthSb->value();
+    settings.useMM        = ui->useMM->isChecked();
+    settings.movingSource = ui->movingSourceRb->isChecked();
+    settings.knifeOnLeft  = ui->knifeOnLeftCb->isChecked();
+    settings.clearCenter  = ui->clearCenterCb->isChecked();
+
+    // Pass the class members that aren't driven by UI widgets
+    settings.heightMultiply = this->heightMultiply;
+    settings.lateralOffset  = this->lateralOffset;
+    // Note: outputLambda should be defined/accessible in your scope
+    settings.outputLambda   = 550.0;
+
+    // 2. Call the refactored static engine for both images
+    QImage ronchiImg = generateOpticalTestImage(OpticalTestType::Ronchi, m_wf, settings);
+    QImage foucaultImg = generateOpticalTestImage(OpticalTestType::Foucault, m_wf, settings);
+
+    // Store for potential saving/external access
+    m_foucultQimage = foucaultImg;
+
+    // 3. UI Painting Helper (to avoid duplicating the Painter logic)
+    auto paintAndDisplay = [&](QLabel* label, QImage img, double offsetValue) {
```

Comment:
```suggestion
    auto paintAndDisplay = [&](QLabel* label, const QImage& img, double offsetValue) {
```
if you need a copy of `img`, do it explicitely in the code for readability. If you don't then this will just improve performance by removing an unnecessary copy